pandas/tests/indexes/test_base.py

```diff
@@ -302,138 +290,138 @@ def test_index_ctor_infer_periodindex(self):
-                    Index(np.array([1., 2., 3.], dtype=float), dtype=float),
-                    Index([1, 2, 3], dtype=float),
-                    Index([1., 2., 3.], dtype=float)]:
-            assert isinstance(idx, Float64Index)
-
-        for idx in [Index(np.array([True, False, True], dtype=bool)),
-                    Index([True, False, True]),
-                    Index(np.array([True, False, True], dtype=bool),
-                          dtype=bool),
-                    Index([True, False, True], dtype=bool)]:
-            assert isinstance(idx, Index)
-            assert idx.dtype == object
+    @pytest.mark.parametrize("cast_idx", [True, False])
+    @pytest.mark.parametrize("vals", [
+        [True, False, True], np.array([True, False, True], dtype=bool)
+    ])
+    def test_constructor_dtypes_to_object(self, cast_idx, vals):
+        if cast_idx:
+            idx = Index(vals, dtype=bool)
+        else:
+            idx = Index(vals)
 
-        for idx in [Index(np.array([1, 2, 3], dtype=int), dtype='category'),
-                    Index([1, 2, 3], dtype='category'),
-                    Index(np.array([np_datetime64_compat('2011-01-01'),
-                                    np_datetime64_compat('2011-01-02')]),
-                          dtype='category'),
-                    Index([datetime(2011, 1, 1), datetime(2011, 1, 2)],
-                          dtype='category')]:
-            assert isinstance(idx, CategoricalIndex)
-
-        for idx in [Index(np.array([np_datetime64_compat('2011-01-01'),
-                                    np_datetime64_compat('2011-01-02')])),
-                    Index([datetime(2011, 1, 1), datetime(2011, 1, 2)])]:
-            assert isinstance(idx, DatetimeIndex)
+        assert isinstance(idx, Index)
+        assert idx.dtype == object
 
-        for idx in [Index(np.array([np_datetime64_compat('2011-01-01'),
-                                    np_datetime64_compat('2011-01-02')]),
-                          dtype=object),
-                    Index([datetime(2011, 1, 1),
-                           datetime(2011, 1, 2)], dtype=object)]:
-            assert not isinstance(idx, DatetimeIndex)
-            assert isinstance(idx, Index)
-            assert idx.dtype == object
+    @pytest.mark.parametrize("vals", [
+        [1, 2, 3], np.array([1, 2, 3], dtype=int),
+        np.array([np_datetime64_compat('2011-01-01'),
+                  np_datetime64_compat('2011-01-02')]),
+        [datetime(2011, 1, 1), datetime(2011, 1, 2)]
+    ])
+    def test_constructor_dtypes_to_categorical(self, vals):
+        idx = Index(vals, dtype='category')
+        assert isinstance(idx, CategoricalIndex)
 
-        for idx in [Index(np.array([np.timedelta64(1, 'D'), np.timedelta64(
-                1, 'D')])), Index([timedelta(1), timedelta(1)])]:
-            assert isinstance(idx, TimedeltaIndex)
+    @pytest.mark.parametrize("cast_idx", [True, False])
```

Comment:
I've used this in a few tests and am not a huge fan of it, but I wasn't sure of a better way to emulate the existing behavior where `dtype=object` is in the `Index` constructor in half of the tests